### PR TITLE
Project-Mongo-API (Elaine B.)

### PR DESCRIPTION
**Pull request by CoffeeDoodles**
https://mlp-mongo.herokuapp.com/

### CODE REVIEW

**Review by Sofarz — APPROVED** [review]
Good job on this weeks project Elaine! You have really put effort into creating your own data collection, and way to go on practicing establishing relations between documents in different collections. However don't forget to add .populate where it is needed :) 
You have reached the blue requirements and if you try out some of my comments, maybe that could solve some of your problems? (I hope so!!) 😚

File: server.js [hunk 1]
```diff
@@ -1,39 +1,165 @@
...
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+//middlewares
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
+const mongoURL = process.env.MONGO_URL || 'mongodb://localhost/myLittlePonyDatabase'
```
[the remark this answers is not in the document]
Comment (CoffeeDoodles) [reply server.js:92]: Oh lord... how... how did I miss that?!

File: server.js [hunk 2]
```diff
@@ -1,39 +1,165 @@
...
+});
+
+
+//Route to all character Data
+app.get('/characters', async (req, res) => {
+  const { name } = req.query;
+  const { kind } = req.query;
+  // const { residence } = req.query;
+  
+  if (name) {
```
[the remark this answers is not in the document]
Comment (CoffeeDoodles) [reply server.js:103]: I've been fiddling with some on the side but have not figured out how to make it work. I'll let you know if I figure it out though!

File: server.js [hunk 3]
```diff
@@ -1,39 +1,165 @@
...
-})
+  res.send(listEndpoints(app))
+});
+
+
+//Route to all character Data
+app.get('/characters', async (req, res) => {
+  const { name } = req.query;
+  const { kind } = req.query;
+  // const { residence } = req.query;
```
[the remark this answers is not in the document]
Comment (CoffeeDoodles) [reply server.js]: Ah I forgot we could also do it that way... much cleaner.

File: server.js [hunk 4]
```diff
@@ -1,39 +1,165 @@
...
+    res.json(characters);
+  }
+});
+
+//Route to charactersID path
+app.get('/characters/:characterId', async (req, res) => {
+  const { characterId } = req.params;
+
+  try {
+    const singleCharacter = await Character.find({ _id: characterId });
```
[the remark this answers is not in the document]
Comment (CoffeeDoodles) [reply server.js]: -facepalm- I can't believe I forgot these...

**Review by KarolCislowski — APPROVED** [review]
Nice API but...
```
"name": "Twilight Sparkle",
    "kind": {
      "_id": "60aac4c40087382eff00cff3",
      "description": "Unicorn",
```
Twilight Sparkle is not a unicorn 😭

**Review by KarolCislowski — APPROVED** [review]
Great API 🦄

**Comment by CoffeeDoodles** [comment]
> Nice API but...
> 
> ```
> "name": "Twilight Sparkle",
>     "kind": {
>       "_id": "60aac4c40087382eff00cff3",
>       "description": "Unicorn",
> ```
> 
> Twilight Sparkle is not a unicorn 😭

Spoilers Karol! SPOILERS! 😂

**Pull request closed without merging**